x-pack/plugins/security/public/management/management_service.ts

```diff
@@ -66,53 +67,57 @@ export class ManagementService {
   }
 
   start({ capabilities, uiConfig }: StartParams) {
-    this.licenseFeaturesSubscription = this.license.features$.subscribe(async (features) => {
-      const securitySection = this.securitySection!;
-
-      const securityManagementAppsStatuses: Array<[ManagementApp, boolean]> = [
-        [securitySection.getApp(apiKeysManagementApp.id)!, features.showLinks],
-      ];
-
-      if (!uiConfig || uiConfig.userManagementEnabled) {
-        securityManagementAppsStatuses.push([
-          securitySection.getApp(usersManagementApp.id)!,
-          features.showLinks,
-        ]);
-      }
-
-      if (!uiConfig || uiConfig.roleManagementEnabled) {
-        securityManagementAppsStatuses.push([
-          securitySection.getApp(rolesManagementApp.id)!,
-          features.showLinks,
-        ]);
-      }
-
-      if (!uiConfig || uiConfig.roleMappingManagementEnabled) {
-        securityManagementAppsStatuses.push([
-          securitySection.getApp(roleMappingsManagementApp.id)!,
-          features.showLinks && features.showRoleMappingsManagement,
-        ]);
-      }
-
-      // Iterate over all registered apps and update their enable status depending on the available
-      // license features.
-      for (const [app, enableStatus] of securityManagementAppsStatuses) {
-        if (capabilities.management.security[app.id] !== true) {
-          app.disable();
-          continue;
-        }
-
-        if (app.enabled === enableStatus) {
-          continue;
-        }
-
-        if (enableStatus) {
-          app.enable();
-        } else {
-          app.disable();
-        }
-      }
-    });
+    this.licenseFeaturesSubscription = this.license.features$
+      .pipe(
+        mergeMap(async (features) => {
```

Comment:
nit: this probably doesn't require an `async` as it seems like the code is synchronous.

Reply:
Addressed in [`0fab35e` (#169522)](https://github.com/elastic/kibana/pull/169522/commits/0fab35e43a6392ec41e358d0cd95140323fac58a)